hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java

```diff
@@ -267,8 +267,7 @@ public TransactionContext startTransaction(
   @Override
   public TransactionContext preAppendTransaction(TransactionContext trx)
       throws IOException {
-    OMRequest request = OMRatisHelper.convertByteStringToOMRequest(
-        trx.getStateMachineLogEntry().getLogData());
+    final OMRequest request = (OMRequest) trx.getStateMachineContext();
     OzoneManagerProtocolProtos.Type cmdType = request.getCmdType();
 
```

Comment:
All these validations should be on `startTransaction` as well.  The creation of UGI is quite expensive as well.

Reply:
The UGI is only for upgrade `Prepare`.  Let's don't optimize it here.